src/models/propulsion/FGBldc.cpp

```diff
@@ -0,0 +1,224 @@
+ Corrected   : 26/12/2021
+ Purpose:      This module models an DLDC electric motor
+
+ --------- Copyright (C) 2020  Matt Vacanti / Paolo Becchi-------------
+
+UPDATE LICENSE
+
+FUNCTIONAL DESCRIPTION
+--------------------------------------------------------------------------------
+
+UPDATE FUNCTIONAL DESCRIPTION
+
+HISTORY
+--------------------------------------------------------------------------------
+11/08/2020  MDV  Created
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+INCLUDES
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+#include <iostream>
+#include <sstream>
+#include <math.h>
+
+#include "FGFDMExec.h"
+#include "FGBldc.h"
+#include "FGPropeller.h"
+#include "input_output/FGXMLElement.h"
+
+using namespace std;
+
+namespace JSBSim {
+
+/*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+CLASS IMPLEMENTATION
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+FGBldc::FGBldc(FGFDMExec* exec, Element *el, int engine_number, struct FGEngine::Inputs& input)
+  : FGEngine(engine_number, input)
+{
+      Load(exec,el);
+
+      Type = etElectric;
+      PowerWatts = 745.7;
+      hptowatts = 745.7;
+
+    // this property is not necessary since is computed using other properties
+    // if (el->FindElement("maxcurrent"))
+    // MaxCurrent= el->FindElementValueAsNumber("maxcurrent");
+
+      if (el->FindElement("maxvolts"))
+        MaxVolts= el->FindElementValueAsNumber("maxvolts");
+
+      if (el->FindElement("velocityconstant"))
+        VelocityConstant= el->FindElementValueAsNumber("velocityconstant");
+
+      if (el->FindElement("torqueconstant"))
+        TorqueConstant= el->FindElementValueAsNumber("torqueconstant");
+  
+      if (el->FindElement("coilresistance"))
+          coilResistance = el->FindElementValueAsNumber("coilresistance");
+      if (el->FindElement("noloadcurrent"))
+          noLoadCurrent = el->FindElementValueAsNumber("noloadcurrent");
+      if (el->FindElement("decelerationTime"))
+          deceleration_time = el->FindElementValueAsNumber("decelerationTime");
+ 
+      MaxCurrent = MaxVolts / coilResistance + noLoadCurrent;
+
+      PowerWatts = MaxCurrent * MaxVolts;
+
+      string base_property_name = CreateIndexedPropertyName("propulsion/engine",
+                                                            EngineNumber);
+      exec->GetPropertyManager()->Tie(base_property_name + "/power-hp", &HP);
+
+      exec->GetPropertyManager()->Tie(base_property_name + "/current-a", &CurrentRequired);
+
+      Debug(0); // Call Debug() routine from constructor if needed
+}
+
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+FGBldc::~FGBldc()
+    {
+      Debug(1); // Call Debug() routine from constructor if needed
+}
+
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+void FGBldc::Calculate(void)
```

Comment:
```suggestion
void FGBrushlessDCMotor::Calculate(void)
```
Explicit class name.